R/zchunk_LA1011.en_bal_adj.R

```diff
@@ -1,88 +1,214 @@
+
+    # This is also complicated. In regions with very low natural gas use and high coal-to-gas with very high input-output coefs on coal-to-gas production
+    # (South Africa), dividing the coal input by the IO coef may cause gas production in excess of the demands in the region.
+    # If this is the case, need to return to original energy balance data and reduce the coal input to gas works (can re-allocate to another sector if desired)
+
+    if(nrow(filter(L1011.en_bal_EJ_R_Si_Fi_Yh, sector == "TPES", fuel == "gas", value < 0)) > 0) {
+      stop("Exogenous IO coef on coal input to gas works caused an increase in natural gas beyond the regional TPES of gas")
+    }
+
+
     # ===================================================
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      add_title("Energy balances by GCAM region / intermediate sector / intermediate fuel / historical year") %>%
+      add_units("EJ") %>%
+      add_comments("Replacing international shipping estimates on IEA energy balances with EIA,") %>%
+      add_comments("subtract coal-to-gas from natural gas TPES") %>%
       add_legacy_name("L1011.en_bal_EJ_R_Si_Fi_Yh") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
+      add_precursors("L101.en_bal_EJ_R_Si_Fi_Yh_full", "energy/A22.globaltech_coef", "energy/calibrated_techs") %>%
       add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
       L1011.en_bal_EJ_R_Si_Fi_Yh
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+
+     L1011.in_EJ_ctry_intlship_TOT_Yh %>%
+      add_title("Liquid fuel consumption by international shipping by country / historical year") %>%
+      add_units("EJ") %>%
+      add_comments("EIA international shipping data converted to EJ, aggregated by country,") %>%
+      add_comments("adding USSR data to Russia") %>%
       add_legacy_name("L1011.in_EJ_ctry_intlship_TOT_Yh") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
+      add_precursors("energy/EIA_RFO_intlship_kbbld", "energy/EIA_TOT_intlship_kbbld", "energy/mappings/EIA_ctry", "common/iso_GCAM_regID") %>%
       add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
       L1011.in_EJ_ctry_intlship_TOT_Yh
 
```

Comment:
@bpbond not sure why this was included in the PR or how to exclude it. Let me know if you have any advice!